server/helpers/adapters/mysql.ts

```diff
@@ -4,40 +4,54 @@ import fleek from '@fleekhq/fleek-storage-js';
 import { isAddress, getAddress } from '@ethersproject/address';
 import db from '../mysql';
 
-export async function storeProposal(space, body, authorIpfsHash, relayerIpfsHash) {
+export async function storeProposal(
+  space,
+  body,
+  authorIpfsHash,
+  relayerIpfsHash
+) {
   const msg = JSON.parse(body.msg);
   let query = 'INSERT IGNORE INTO messages SET ?;';
-  await db.queryAsync(query, [{
-    id: authorIpfsHash,
-    address: body.address,
-    version: msg.version,
-    timestamp: msg.timestamp,
-    space,
-    type: 'proposal',
-    payload: JSON.stringify(msg.payload),
-    sig: body.sig,
-    metadata: JSON.stringify({
-      relayer_ipfs_hash: relayerIpfsHash
-    })
-  }]);
+  await db.queryAsync(query, [
+    {
+      id: authorIpfsHash,
+      address: body.address,
+      version: msg.version,
+      timestamp: msg.timestamp,
+      space,
+      type: 'proposal',
+      payload: JSON.stringify(msg.payload),
+      sig: body.sig,
+      metadata: JSON.stringify({
+        relayer_ipfs_hash: relayerIpfsHash
+      })
+    }
+  ]);
+}
+
+export async function archiveProposal(authorIpfsHash) {
+  let query = 'UPDATE messages SET type = ? WHERE id = ?';
```

Comment:
It's better to add a LIMIT here
```suggestion
  let query = 'UPDATE messages SET type = ? WHERE id = ? LIMIT 1';
```
Also it's better to do a WHERE type = 'proposal'